aiohttp_session/cookie_storage.py

```diff
@@ -33,3 +37,54 @@ def save_session(self, request, response):
 
         cookie_data = json.dumps(session._mapping)
         self.store_cookie(response, cookie_data)
+
+
+class EncryptedCookieStorage(AbstractStorage):
+    """Encrypted JSON storage.
+    """
+
+    def __init__(self, secret_key, identity="AIOHTTP_COOKIE_SESSION", *,
+                 domain=None, max_age=None, path='/',
+                 secure=None, httponly=True, iv=None):
+        super().__init__(identity, domain=domain, max_age=max_age,
+                         path=path, secure=secure, httponly=httponly)
+
+        self.__secret_key = secret_key
+        if len(self.__secret_key) % AES.block_size != 0:
+            raise TypeError(
+                'Secret key must be a multiple of {} in length'.format(
+                    AES.block_size))
+        self.__iv = iv
```

Comment:
Initial vector should be never passed to storage constructor nor stored in storage attr. It should be stored in cookie together with encoded data.